AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine+Retryable.swift

```diff
@@ -0,0 +1,64 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+
+@available(iOS 13.0, *)
+extension RemoteSyncEngine {
+
+    func resetCurrentAttemptNumber() {
+        currentAttemptNumber = 1
+    }
+
+    func scheduleRestart(error: AmplifyError?) {
+        let advice = getRetryAdviceIfRetryable(error: error)
+        if advice.shouldRetry {
+            scheduleRestart(advice: advice)
+        } else {
+            if let error = error {
+                remoteSyncTopicPublisher.send(completion: .failure(DataStoreError.api(error)))
+            } else {
+                remoteSyncTopicPublisher.send(completion: .finished)
+            }
+        }
+
+    }
+
+    private func getRetryAdviceIfRetryable(error: Error?) -> RequestRetryAdvice {
```

Comment:
nit: "ifRetryable" is redundant, since the advice could be "don't retry".

Reply:
Updated, thanks!

Reply:
addressed: https://github.com/aws-amplify/amplify-ios/pull/329